README.md

```diff
@@ -2,30 +2,79 @@
+4. The string contains at least one non-digit.
+
 
 ## Instructions
 
-Walk the student through any setup required to run the lesson (i.e.
-`npm install` and `npm start`).
+The starter project has a class `PasswordChecker` with several stub methods that return
+a default value.  You will adapt the methods to implement the password checker based on the
+rules above.   You are also provided a Junit test class `PasswordCheckerTest`, which
+has the test methods disabled.  As you solve each task, remove the `@Disabled` annotation
+to run the test method and confirm your implementation.
+
+Do not use regular expressions to solve any of the tasks.
+
+You may use the methods of the [Java 17 String](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/String.html) class.
+
+You may adapt the example string utility methods from the lessons.
+
+Do not import string utility classes from external packages.
+
+## Task #1
 
-## Deliverables
+1. The `PasswordChecker` class contains an `isValidLength` method that returns `false`.  
+   Edit the method to return a boolean indicating if the parameter string length is at least 8.
+2. Edit `PasswordCheckerTest` to remove the `@Disabled` annotation from the `isValidLength` method.
+3. Run `PasswordCheckerTest` and confirm the test passes.
 
-List each of the deliverables the student must complete in order to finish the
-lab. Provide as much context as possible, including instructions on how to run
-the tests and other means of validating successful completion of deliverables.
+## Task #2
 
-## Conclusion
+1. The `PasswordChecker` class contains a `containsSpecialChar` method that returns `false`.   
+   Edit the method to return a boolean indicating if the parameter string contains at least one
+   of the special characters '!' or '#' or '?'.
+2. Edit `PasswordCheckerTest` to remove the `@Disabled` annotation from the `containsSpecialChar` method.
+3. Run `PasswordCheckerTest` and confirm the test passes.
+
+## Task #3
+
+1. The `PasswordChecker` class contains a `containsDigit` method that returns `false`.   
+   Edit the method to return a boolean indicating if the parameter string contains at least one digit
+   (HINT: test for characters within a range).
+2. Edit `PasswordCheckerTest` to remove the `@Disabled` annotation from the `containsDigit` method.
+3. Run `PasswordCheckerTest` and confirm the test passes.
+
+## Task #4
+
+1. The `PasswordChecker` class contains a `containsNonDigit` method that returns `false`.   
+   Edit the method to return a boolean indicating if the parameter string contains at least
+   character that is **not** a digit (HINT: test for characters outside a range).
+2. Edit `PasswordCheckerTest` to remove the `@Disabled` annotation from the `containsNonDigit` method.
+3. Run `PasswordCheckerTest` and confirm the test passes.
+
+## Task #5
+
+1. The `PasswordChecker` class contains a `passwordStrength` method that returns `0`.   
```

Comment:
I know they can see this by looking at the unit test, but maybe here provide an example of exactly that?

"If all four of the methods from the previous tasks return `true`, then the password strength should be 4." or something along those lines to clarify more?

Reply:
ok, added example text.